src/pkgconf/_path_entrypoints.py

```diff
@@ -0,0 +1,213 @@
+import contextlib
+import importlib.machinery
+import importlib.metadata
+import importlib.resources
+import importlib.util
+import operator
+import os
+import pathlib
+import pickle
+import sys
+import textwrap
+import types
+import warnings
+
+from collections.abc import Callable, Iterable
+from typing import Any, ParamSpec, TypeVar
+
+import pkgconf
+
+
+P = ParamSpec('P')
+T = TypeVar('T')
+
+
+# Import system helpers
+
+
+def import_module_no_exec(name: str) -> types.ModuleType:
+    """Return a module object with all the module attributes set, but without executing it.
+
+    WARNING: This places uninitialized modules in sys.modules.
+    """
+    if name in sys.modules:
+        return sys.modules[name]
+    # Import parent without execution
+    if parent := name.rpartition('.')[0]:
+        import_module_no_exec(parent)
+    # Find module spec
+    spec = importlib.util.find_spec(name)
+    if not spec:
+        msg = f'No module named {name!r}'
+        raise ModuleNotFoundError(msg)
+    # Create module object without executing, and save it sys.modules
+    module = sys.modules[name] = importlib.util.module_from_spec(spec)
+    return module
+
+
+def module_path(name: str) -> str:
+    """Resolve module name to file-system path.
+
+    WARNING: This places uninitialized modules in sys.modules.
+    """
+    module = import_module_no_exec(name)
+    # Check if submodule_search_locations is set on the spec.
+    if module.__spec__.submodule_search_locations:
+        # If it contains a single path, use it.
+        if len(module.__spec__.submodule_search_locations) == 1:
+            return module.__spec__.submodule_search_locations[0]
+    # Traversables often implement __fspath__, attempt to use it.
+    traversable = importlib.resources.files(module)
+    if isinstance(traversable, os.PathLike):
+        return os.fsdecode(os.fspath(traversable))
+    # Give up :/
+    msg = f'Unable to resolve the path for {name}'
+    raise ValueError(msg)
+
+
+# Helpers to run the import helpers isolated from the import state of the main process/interpreter
+
+
+def run_in_subinterpreter(fn: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> T:
+    """Run callable in a subinterpreter — using concurrent.interpreters."""
+    assert sys.version_info >= (3, 14)
+
+    import concurrent.interpreters
```

Comment:
I'm sure you tested it, but probably should add 3.14 to CI just to make sure?

Reply:
Ah, yes. Thanks!